bindings/rust/s2n-tls/src/config.rs

```diff
@@ -595,6 +635,38 @@ impl Builder {
         Ok(self)
     }
 
+    /// Enable negotiating session tickets in a TLS connection
+    pub fn enable_session_tickets(&mut self) -> Result<&mut Self, Error> {
+        unsafe { s2n_config_set_session_tickets_onoff(self.as_mut_ptr(), 1).into_result() }?;
+        Ok(self)
+    }
+
+    /// Adds a key which will be used to encrypt and decrypt session tickets
+    pub fn add_session_ticket_key(
+        &mut self,
+        key_name: &[u8],
+        key: &[u8],
+        intro_time: u64,
+    ) -> Result<&mut Self, Error> {
+        let key_name_len: u32 = key_name
+            .len()
+            .try_into()
+            .map_err(|_| Error::INVALID_INPUT)?;
+        let key_len: u32 = key.len().try_into().map_err(|_| Error::INVALID_INPUT)?;
+        unsafe {
+            s2n_config_add_ticket_crypto_key(
+                self.as_mut_ptr(),
+                key_name.as_ptr(),
+                key_name_len,
+                key.as_ptr() as *mut u8,
```

Comment:
Yeah this really should have been `const` on the C side... We should look into if it would be a breaking change to change that.

Reply:
If you change it to be const, than users won't be able to build unless they update their parameter type...Doesn't that count as a breaking change?

Reply:
Seems ok on the C side of things? https://godbolt.org/z/EM3KaTM36

Reply:
Same with rust: https://godbolt.org/z/n5cqjcfrW

Reply:
If we change the parameter to be const then this line will error because our blob functions don't work on pointers to const: https://github.com/aws/s2n-tls/blob/main/tls/s2n_config.c#L907. So we'd have to add a memcopy to get it into the blob.

Reply:
Can you open an issue for this? It seems backwards to have every caller have to work around blob limitations, right?

Reply:
#4140